### PR TITLE
Measure isBlocked and isFinished operator calls

### PR DESCRIPTION
**Pull request by yingsu00**

### CODE REVIEW

**Comment by yingsu00** [comment]
@Yuhta @xiaoxmeng Can you please review this? Thanks!

File: velox/exec/tests/TaskTest.cpp [hunk 1]
```diff
@@ -1184,7 +1184,7 @@ DEBUG_ONLY_TEST_F(TaskTest, liveStats) {
   EXPECT_EQ(32 * numBatches, operatorStats.outputBytes);
   EXPECT_EQ(3 * numBatches, operatorStats.outputPositions);
   EXPECT_EQ(numBatches, operatorStats.outputVectors);
-  EXPECT_EQ(1, operatorStats.finishTiming.count);
+  EXPECT_EQ(2, operatorStats.finishTiming.count);
```
Comment (Yuhta) [inline velox/exec/tests/TaskTest.cpp:1187]: Is there a way to check `isBlockedTiming` e.g. in HashProbe?

Comment (yingsu00) [reply velox/exec/tests/TaskTest.cpp:1189]: yes. I just added it to this test.
```
for (auto i = 0; i < numBatches; ++i) {
    ...
    EXPECT_EQ(2 * (i + 1), operatorStats.isBlockedTiming.count);
```
Sorry I didn't add it until now. I was blocked by the build issues for a long time.

**Review by xiaoxmeng — COMMENTED** [review]
@yingsu00 LGTM and can you complete the test? Thanks!

**Review by xiaoxmeng — APPROVED** [review]
@yingsu00 thanks!

File: velox/exec/tests/TaskTest.cpp [hunk 2]
```diff
@@ -1184,7 +1185,8 @@ DEBUG_ONLY_TEST_F(TaskTest, liveStats) {
   EXPECT_EQ(32 * numBatches, operatorStats.outputBytes);
   EXPECT_EQ(3 * numBatches, operatorStats.outputPositions);
   EXPECT_EQ(numBatches, operatorStats.outputVectors);
-  EXPECT_EQ(1, operatorStats.finishTiming.count);
+  EXPECT_EQ(2 * numBatches, operatorStats.isBlockedTiming.count);
```
Comment (kevinwilfong) [inline velox/exec/tests/TaskTest.cpp:1188]: It looks like this condition is failing in the tests

File: velox/exec/Driver.cpp [hunk 3]
```diff
@@ -556,23 +556,28 @@ StopReason Driver::runInternal(
...
 
         if (i < numOperators - 1) {
           Operator* nextOp = operators_[i + 1].get();
 
-          CALL_OPERATOR(
-              blockingReason_ = nextOp->isBlocked(&future),
-              nextOp,
-              curOperatorId_ + 1,
-              kOpMethodIsBlocked);
+          withDeltaCpuWallTimer(op, &OperatorStats::isBlockedTiming, [&]() {
```
Comment (lingbin) [inline velox/exec/Driver.cpp:574]: Should be `nextOp` here, The time consumed by `nextOp->isBlocked()` counted here should be accumulated to `nextOp`, not `op`.

See https://github.com/facebookincubator/velox/issues/12173

Comment (Chiassonastraypatrick) [reply velox/exec/Driver.cpp:574]: Will see this might take some time

**Review by Chiassonastraypatrick — APPROVED** [review]

**Review by Chiassonastraypatrick — APPROVED** [review]

**Comment by Real-Chen-Happy** [comment]
Hi @yingsu00 , may I know is there any cases you have found CPU metrics largely off because of not collecting isBlocked and isFinished?

The reasons for asking is that we have found collecting these additional CPU metrics have some regressions over our benchmark (For Gluten Spark). We want to see the impact of this commit. Thanks.

**Pull request closed without merging**